go/flowctl/cmd-api-activate.go

```diff
@@ -0,0 +1,308 @@
+	"github.com/estuary/flow/go/labels"
+	"github.com/estuary/flow/go/materialize"
+	"github.com/estuary/protocols/catalog"
+	pf "github.com/estuary/protocols/flow"
+	pm "github.com/estuary/protocols/materialize"
+	log "github.com/sirupsen/logrus"
+	"go.gazette.dev/core/broker/client"
+	pb "go.gazette.dev/core/broker/protocol"
+	"go.gazette.dev/core/consumer"
+	pc "go.gazette.dev/core/consumer/protocol"
+	mbp "go.gazette.dev/core/mainboilerplate"
+)
+
+type apiActivate struct {
+	All            bool                  `long:"all" description:"Activate all tasks and collections"`
+	AllDerivations bool                  `long:"all-derivations" description:"Activate all derivations"`
+	Broker         mbp.ClientConfig      `group:"Broker" namespace:"broker" env-namespace:"BROKER"`
+	BuildID        string                `long:"build-id" required:"true" description:"ID of this build"`
+	BuildsRoot     string                `long:"builds-root" required:"true" env:"BUILDS_ROOT" description:"Base URL for fetching Flow catalog builds"`
+	Consumer       mbp.ClientConfig      `group:"Consumer" namespace:"consumer" env-namespace:"CONSUMER"`
+	DryRun         bool                  `long:"dry-run" description:"Print actions that would be taken, but don't actually take them"`
+	InitialSplits  int                   `long:"initial-splits" default:"1" description:"When creating new tasks, the number of initial key splits to use"`
+	Names          []string              `long:"name" description:"Name of task or collection to activate. May be repeated many times"`
+	Network        string                `long:"network" default:"host" description:"The Docker network that connector containers are given access to."`
+	Log            mbp.LogConfig         `group:"Logging" namespace:"log" env-namespace:"LOG"`
+	Diagnostics    mbp.DiagnosticsConfig `group:"Debug" namespace:"debug" env-namespace:"DEBUG"`
+}
+
+func (cmd apiActivate) execute(ctx context.Context) error {
+	var builds, err = flow.NewBuildService(cmd.BuildsRoot)
+	if err != nil {
+		return err
+	}
+
+	ctx = pb.WithDispatchDefault(ctx)
+	var sc = cmd.Consumer.MustShardClient(ctx)
+	var jc = cmd.Broker.MustJournalClient(ctx)
+
+	// Ping to ensure connectivity.
+	if err = pingClients(ctx, sc, jc); err != nil {
+		return err
+	}
+
+	var build = builds.Open(cmd.BuildID)
+	defer build.Close()
+
+	// Identify collections and tasks of the build to activate.
+	var collections []*pf.CollectionSpec
+	var tasks []pf.Task
+
+	if err := build.Extract(func(db *sql.DB) error {
+
+		// Index explicit names to activate. If --all or --all-derivations is set,
+		// additional names not in |names| may also be activated.
+		var names = make(map[string]struct{})
+		for _, t := range cmd.Names {
+			names[t] = struct{}{}
+		}
+
+		if all, err := catalog.LoadAllCollections(db); err != nil {
+			return err
+		} else {
+			for _, c := range all {
+				var _, ok = names[c.Collection.String()]
+				if ok || cmd.All || cmd.AllDerivations {
```

Comment:
I was curious about what would happen if I pass the `--name` of a capture, but not the name of the collection that it's capturing into. If you build `examples/local-sqlite.flow.yaml` and then run `flowctl api activate --name examples/source-hello-world` (other args elided), it succeeds and creates the capture shard, but the capture doesn't seem to actually run, beyond just initializing the task term. It doesn't seem to log anything about the missing collection, either.

I'm not really sure what the _correct_ behavior would be, though. Error during `activate`? Automatically activate the collections named in the bindings of captures and materializations? Always activate all collections?